src/auth/auth.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { RegisterDto } from './dto/register.dto';
+import { UserService } from 'src/user/user.service';
+import { RefreshDto } from './dto/refresh.dto';
+import {
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common/exceptions';
+import { verify } from 'argon2';
+import { LoginDto } from './dto/login.dto';
+@Injectable()
+export class AuthService {
+  constructor(
+    private jwtService: JwtService,
+    private userService: UserService,
+  ) {}
+  async register(registerDto: RegisterDto) {
+    const user = await this.userService.create(registerDto);
+
+    const tokens = await this.issueTokens(user.id);
+
+    return {
+      user,
+      ...tokens,
+    };
+  }
+
+  async refresh(refreshDto: RefreshDto) {
+    const result = await this.jwtService.verifyAsync(refreshDto.refreshToken);
+
+    if (!result) {
+      throw new UnauthorizedException('Invalid refresh token');
+    }
+
+    const user = await this.userService.findUnique(result.id);
+
+    const tokens = await this.issueTokens(user.id);
+
+    return { ...tokens };
+  }
+
+  async login(loginDto: LoginDto) {
+    const user = await this.validateUser(loginDto);
+    const tokens = await this.issueTokens(user.id);
+
+    return {
+      user: {
+        id: user.id,
+        name: user.name,
+        email: user.email,
+      },
+      ...tokens,
+      jwtExpiresIn: new Date().setTime(new Date().getTime() + 20 * 1000),
```

Comment:
I suggest you to install some date manipulating package (for example `date-fns`)

```ts
addSeconds(new Date(), 20)
```

Also, you should somehow move this amount of time to a constant. For example.

`auth.constants.ts`
```ts
export const JWT_EXPIRES_IN_SECONDS = process.env.JWT_EXPIRES_IN_SECONDS
```

...
```ts
jwtExpiresIn: addSeconds(new Date(), JWT_EXPIRES_IN_SECONDS)
```
Also, you can reuse it in `auth.module.ts` in `JWTModule.register`